### PR TITLE
Fix PGO Pipeline

### PR DESCRIPTION
**Pull request by kmahone**
The existing PGO versioning logic hit an issue at the start of this year, since it used a YYMMDDhhmm format for part of the version of the nuget package which hit an overflow issue once the year became "22".

This is primarily a port of the corresponding fix from WinUI3.

### CODE REVIEW

File: tools/MUXPGODatabase/restore-pgodb.ps1 [hunk 1]
```diff
@@ -1,47 +1,89 @@
...
 
-Write-Host ( "PGO OPTIMIZE: requesting {0} version {1}" -f $packageId, ( FormatVersion $currentVersion ) )
+    foreach ( $line in $output )
+    {
+        $name, $version = $line.Split(" ")
 
-$packageSource = Register-PackageSource -ForceBootstrap -Name MUX_Dependencies -Location $feedUri -ProviderName NuGet -Trusted
-$packages = ( Find-Package $packageId -Source MUX_Dependencies -AllowPrereleaseVersions -AllVersions ) | Sort-Object -Property Version -Descending
+        # Some packages on the feed use an old versioning scheme. Skip the package if it does not include a -foo portion in the version.
+        if($version.Contains("-"))
```
Comment (bartekk8) [inline tools/MUXPGODatabase/restore-pgodb.ps1:26]: Why is this necessary?  This should be taken care of when version obj is created.

Comment (kmahone) [reply tools/MUXPGODatabase/restore-pgodb.ps1:26]: It was tripping up on some package versions with versions like "2.0.0" - I'm not sure where those came from - probably from when PGO was first getting set up.

Comment (kmahone) [reply tools/MUXPGODatabase/restore-pgodb.ps1]: Good idea. I've delisted the packages and removed this condition.
